base/multidimensional.jl

```diff
@@ -1565,8 +1565,8 @@ end
     end
 end
 
-isassigned(a::AbstractArray, i::CartesianIndex) = isassigned(a, Tuple(i)...)
-function isassigned(A::AbstractArray, i::Union{Integer, CartesianIndex}...)
+@propagate_inbounds isassigned(a::AbstractArray, i::CartesianIndex) = isassigned(a, Tuple(i)...)
+@propagate_inbounds function isassigned(A::AbstractArray, i::Union{Integer, CartesianIndex}...)
```

Comment:
```suggestion
@inline function isassigned(A::AbstractArray, i::Union{Integer, CartesianIndex}...)
```

Reply:
Wouldn't the `@inbounds` need to be propagated to the `isassigned` call on line 1570?

Reply:
Ugh, yes, I didn't see that usage.  I'd still rather more finely scope this.  Maybe split the method instead?

Reply:
Updated